demo/pyproject.toml

```diff
@@ -0,0 +1,106 @@
+[tool.hatch.version]
+path = "src/__init__.py"
+allow-direct-references = true
+
+[project]
+name = 'demo'
+description = 'FastUI demo project'
+authors = [{ name = 'Samuel Colvin', email = 's@muelcolvin.com' }]
+classifiers = [
+    'Development Status :: 5 - Production/Stable',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: Implementation :: CPython',
+    'Programming Language :: Python :: Implementation :: PyPy',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3 :: Only',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
+    'Intended Audience :: Developers',
+    'Intended Audience :: Information Technology',
+    'Intended Audience :: System Administrators',
+    'Operating System :: Unix',
+    'Operating System :: POSIX :: Linux',
+    'Environment :: Console',
+    'Environment :: MacOS X',
+    'Framework :: Hypothesis',
+    'Framework :: Pydantic',
+    'Topic :: Software Development :: Libraries :: Python Modules',
+    'Topic :: Internet',
+]
+requires-python = '>=3.8'
+dependencies = [
+    "fastapi>=0.110.2",
+    "httpx>=0.27.0",
+    "openai==1.10.0",
+    "PYjwt>=2.8.0",
+    "pydantic[email]>=2.7.1",
+]
+dynamic = ['version']
+
+
+[tool.ruff]
+line-length = 120
+target-version = 'py38'
+lint.extend-select = ['Q', 'RUF100', 'C90', 'UP', 'I', 'D', 'T']
+lint.extend-ignore = [
+    'D100',
+    'D101',
+    'D102',
+    'D103',
+    'D104',
+    'D105',
+    'D107',
+    'D205',
+    'D415',
+    'T201',
+]
+lint.flake8-quotes = { inline-quotes = 'single', multiline-quotes = 'double' }
+lint.isort = { known-first-party = ['pydantic', 'tests'] }
+lint.mccabe = { max-complexity = 14 }
+lint.pydocstyle = { convention = 'google' }
+
+[tool.ruff.lint.per-file-ignores]
+'docs/*' = ['D']
+'pydantic/__init__.py' = ['F405', 'F403', 'D']
+'tests/test_forward_ref.py' = ['F821']
+'tests/*' = ['D']
+'pydantic/deprecated/*' = ['D']
+'pydantic/json_schema.py' = ['D']
+
+[tool.ruff.lint.extend-per-file-ignores]
+"docs/**/*.py" = ['T']
+"tests/**/*.py" = ['T', 'E721', 'F811']
+"tests/benchmarks/test_fastapi_startup_generics.py" = ['UP006', 'UP007']
+"tests/benchmarks/test_fastapi_startup_simple.py" = ['UP006', 'UP007']
+
+[tool.ruff.format]
+quote-style = 'single'
+
+[tool.coverage.run]
+source = ['pydantic']
```

Comment:
This seems off

Reply:
true my bad, i tried to copy from pydantic's rule, some were dead codes.